workflow-templates/common-deploy.yml

```diff
@@ -0,0 +1,344 @@
+
+      - name: Set env 'DEPLOY_TARGETS' for push (to alpha/staging)
+        if: env.BRANCH_NAME == 'develop' && env.PUSH_DEPLOY_TARGET != 'none'
+        run: echo DEPLOY_TARGETS=$(echo ${PUSH_DEPLOY_TARGET} | sed 's|${COMPONENT_NAME}|'"${{ env.COMPONENT_NAME }}"'|g') >> ${GITHUB_ENV}
+
+      - name: Set env 'DEPLOY_TARGETS' pre-release
+        if: env.IS_PRERELEASE == 'true' && env.PRE_RELEASE_DEPLOY_TARGET != 'none'
+        run: echo DEPLOY_TARGETS=$(echo ${PRE_RELEASE_DEPLOY_TARGET} | sed 's|${COMPONENT_NAME}|'"${{ env.COMPONENT_NAME }}"'|g') >> ${GITHUB_ENV}
+
+      - name: Set env 'DEPLOY_TARGETS' for release
+        if: env.IS_RELEASE == 'true' &&  env.RELEASE_DEPLOY_TARGET != 'none'
+        run: echo DEPLOY_TARGETS=$(echo ${RELEASE_DEPLOY_TARGET} | sed 's|${COMPONENT_NAME}|'"${{ env.COMPONENT_NAME }}"'|g') >> ${GITHUB_ENV}
+
+      - name: Set env 'DEPLOY_TARGETS' for master branch (only run build task; never deploy)
+        if: env.BRANCH_NAME == 'master'
+        run: echo DEPLOY_TARGETS="none" >> ${GITHUB_ENV}
+
+      - name: Log details
+        run: |
+          echo "BRANCH_NAME: ${{ env.BRANCH_NAME }}"
+          echo "COMPONENT_NAME: ${{ env.COMPONENT_NAME }}"
+          echo "DEPLOY_TARGETS: ${{ env.DEPLOY_TARGETS }}"
+          echo "DEPLOYMENT_REPO: ${{ env.DEPLOYMENT_REPO }}"
+          echo "DOCKER_TAG_PREFIX: ${{ env.DOCKER_TAG_PREFIX }}"
+          echo "GITHUB_ACTOR: ${GITHUB_ACTOR}"
+          echo "GITHUB_EVENT_NAME: ${GITHUB_EVENT_NAME}"
+          echo "GITHUB_REPOSITORY: ${GITHUB_REPOSITORY}"
+          echo "GITHUB_REF: ${GITHUB_REF}"
+          echo "GITHUB_SHA: ${GITHUB_SHA}"
+          echo "IS_AUTO_RELEASE: ${{ env.IS_AUTO_RELEASE }}"
+          echo "IS_MANUAL_RELEASE: ${{ env.IS_MANUAL_RELEASE }}"
+          echo "IS_PRERELEASE: ${{ env.IS_PRERELEASE }}"
+          echo "IS_RELEASE: ${{ env.IS_RELEASE }}"
+          echo "TAG: ${{ env.TAG }}"
+          echo "VERSION: ${{ env.VERSION }}"
+
+      - name: Create new pre-release
+        if: env.IS_AUTO_RELEASE == 'true' && env.IS_PRERELEASE == 'true'
+        uses: ncipollo/release-action@v1
+        with:
+          token: "${{ secrets.GITHUB_TOKEN }}"
+          allowUpdates: true
+          draft: false
+          prerelease: true
+
+      - name: Create new release
+        if: env.IS_AUTO_RELEASE == 'true' && env.IS_RELEASE == 'true'
+        uses: ncipollo/release-action@v1
+        with:
+          token: "${{ secrets.GITHUB_TOKEN }}"
+          allowUpdates: true
+          draft: false
+          prerelease: false
+
+      ###
+      #
+      #  Implement your own deploy file from here
+      #
+      ###
+
+      - name: Throw unimplemented workflow exception (delete this step)
+        run: exit 1 "Please implement this deploy.yml file to your needs"
+
+      # - name: Prepare gradle properties
+      #   run: printf "\n$GRADLE_PROPS" >> gradle.properties
+      #   env:
+      #     GRADLE_PROPS: ${{ secrets.GRADLE_PROPERTIES }}
```

Comment:
Comments to give developer an example while implementing manual workflow.

Reply:
I would not do a "own deploy file" thing. I would really like all deploy files to be exactly the same. Otherwise auto-update will be problematic...

Reply:
Maybe IGNORE_WORKFLOWS=['deploy.yml'] in workflow-config.env to ignore specific workflows (that will not be used)